### ISSUE TITLE
Quotes in NewSQLQuery

### ISSUE THREAD

**Issue by ghost** [istreamdata/orientgo#28]
Hi,

Trying to do lookup by song title, but getting an exception on songs with double-quotes.

Command:

``` Go
result, err := orientdb.Sess.Command(orient.NewSQLQuery(`SELECT FROM Song WHERE title=?`, itemName).Limit(1))
```

Exception:

```
OrientDB Server Exception: 
  com.orientechnologies.orient.core.sql.OCommandSQLParsingException: Error on parsing command at position #49: Invalid keyword 'BESAME""'
Command: SELECT FROM Song WHERE title = "Para Que Peliar "Besame""
-----------------------------------------^
OrientDB Server Exception: 
  java.lang.IllegalStateException: Missing closed string character: '"', position: 52
```

Is there a correct way to take of quotes in these cases or is this a bug? 

Thank you!

**Comment by dennwc** [comment]
Hi @oharlem,

As I can see, everything is okay with your example, so it seems to be a bug in driver or database itself.

Can you provide some additional info:
1) OrientDB version
2) `itemName` variable is just `Para Que Peliar "Besame"` (without outer quotes), right?
3) `itemName` variable type is `string`?

This problem seems weird to me, as binary protocol just passes strings directly to OrientDB without substituting it to command string, so I've expected not to deal with escaping on driver side.

**Update:** cannot reproduce on both OrientDB 2.0.15 and 2.1.2.

**Comment by ghost** [comment]
Hi @dennwc !
1. Version: 2.1.0
2. Correct. No outer quotes.
3. Correct. String.

My current workaround is to replace double quotes with their escaped version: " -> \" (which is kind of ugly). For some reason I was expecting binding to take care of this.

Also, interesting you mentioned that you cannot reproduce on 2.0.15. I'm not 100% ыгку, but earlier I had 2.0.14 and I never had this issue then. Unfortunately cannot setup 2.1.2 - I'm also using PHPOrient to access the db from front-end and PHPOrient cannot work with 2.1.2 yet.

Hope this helps

**Comment by dennwc** [comment]
Seems that it is really a bug in Orient 2.1.0, as my test failed on this particular version. Still, I haven't decided yet how to handle it. Both servers advertise the same version of protocol, both uses the same serialization scheme. So, it would be a real hack to check OrientDB version string on handshake and put some deep hook into serialization routine to escape strings for that version. I think it would be better for you to switch to OrientDB 2.1.1 at least, if PHPOrient supports it. I will state that 2.1.0 is not supported in a README.

**Comment by dennwc** [comment]
Also disabling [strict SQL parser](http://orientdb.com/docs/last/Release-2.1.0.html#strict-sql-parser) may help, as it was introduced in 2.1.0 and had some issues, that were fixed in 2.1.1.
